src/Metrics/Transports/OpenTelemetryMeterProviderFactory.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\OpenTelemetry\Metrics\Transports;
+
+use OpenTelemetry\API\Behavior\LogsMessagesTrait;
+use OpenTelemetry\API\Metrics\MeterProviderInterface;
+use OpenTelemetry\SDK\Common\Attribute\Attributes;
+use OpenTelemetry\SDK\Common\Configuration\Configuration;
+use OpenTelemetry\SDK\Common\Configuration\KnownValues;
+use OpenTelemetry\SDK\Common\Configuration\Variables;
+use OpenTelemetry\SDK\Common\Instrumentation\InstrumentationScopeFactory;
+use OpenTelemetry\SDK\Common\Time\ClockFactory;
+use OpenTelemetry\SDK\Common\Util\ShutdownHandler;
+use OpenTelemetry\SDK\Metrics\Aggregation\ExplicitBucketHistogramAggregation;
+use OpenTelemetry\SDK\Metrics\Exemplar\ExemplarFilter\AllExemplarFilter;
+use OpenTelemetry\SDK\Metrics\Exemplar\ExemplarFilter\NoneExemplarFilter;
+use OpenTelemetry\SDK\Metrics\Exemplar\ExemplarFilter\WithSampledTraceExemplarFilter;
+use OpenTelemetry\SDK\Metrics\Exemplar\ExemplarFilterInterface;
+use OpenTelemetry\SDK\Metrics\MeterProvider;
+use OpenTelemetry\SDK\Metrics\MetricExporter\NoopMetricExporter;
+use OpenTelemetry\SDK\Metrics\MetricReader\ExportingReader;
+use OpenTelemetry\SDK\Metrics\NoopMeterProvider;
+use OpenTelemetry\SDK\Metrics\StalenessHandler\NoopStalenessHandlerFactory;
+use OpenTelemetry\SDK\Metrics\View\CriteriaViewRegistry;
+use OpenTelemetry\SDK\Metrics\View\SelectionCriteria\InstrumentNameCriteria;
+use OpenTelemetry\SDK\Metrics\View\ViewTemplate;
+use OpenTelemetry\SDK\Registry;
+use OpenTelemetry\SDK\Resource\ResourceInfoFactory;
+use OpenTelemetry\SDK\Sdk;
+
+/**
+ * @internal
+ */
+class OpenTelemetryMeterProviderFactory
+{
+    use LogsMessagesTrait;
+
+    /**
+     * This method is required to be able to configure custom buckets for histograms (or other adjustments if needed).
+     * It is implemented to the metric provider initialization during OpenTelemetry SDK autoloading.
+     *
+     * It may be required to update this method when the OpenTelemetry SDK is updated.
+     *
+     * References:
+     *   - https://github.com/open-telemetry/opentelemetry-php/blob/main/src/SDK/SdkAutoloader.php
+     *   - https://github.com/open-telemetry/opentelemetry-php/blob/main/src/SDK/Metrics/MeterProviderBuilder.php
+     *   - https://github.com/open-telemetry/opentelemetry-php/blob/main/examples/metrics/basic.php
+     *
+     * @param array<string,array<float|int>> $buckets
+     */
+    public function createMeterProvider(array $buckets): MeterProviderInterface
+    {
+        if (Sdk::isDisabled()) {
+            return new NoopMeterProvider();
+        }
+
+        $exporters = Configuration::getList(Variables::OTEL_METRICS_EXPORTER);
+        // todo: check sdk update for "The SDK MAY accept a comma-separated list to enable setting multiple exporters"
+        if (count($exporters) !== 1) {
+            self::logWarning(sprintf('Configuration %s supports exactly 1 exporter, using first from the list', Variables::OTEL_METRICS_EXPORTER));
```

Comment:
I am using the same logging logic here which SDK uses (and would use in case autoconfiguring metric provider). As we use also tracing, I would say it make sense to run all the logs through the same logger.